src/foam/nanos/auth/UserAndGroupAuthService.java

```diff
@@ -126,13 +145,32 @@ public User login(X x, long userId, String password) throws AuthenticationExcept
       throw new AuthenticationException("Invalid Password");
     }
 
+    // Freeze user
+    user = (User) user.fclone();
+    user.freeze();
+
     Session session = x.get(Session.class);
     session.setUserId(user.getId());
     session.setContext(session.getContext().put("user", user));
     sessionDAO_.put(session);
+ 
     return user;
   }
 
+  /**
+   * Login a user by the id provided, validate the password
+   * and return the user in the context.
+   */
+  public User login(X x, long userId, String password) throws AuthenticationException {
+    if ( userId < 1 || SafetyUtil.isEmpty(password) ) {
+      throw new AuthenticationException("Invalid Parameters");
+    }
+
+    User user = (User) userDAO_.find(userId);
```

Comment:
Please call context-oriented methods when you're in a context-oriented method.

Reply:
I don't believe this is necessary considering this is an interface service not a decorator. I could be wrong. @kgrgreer

Reply:
With regard to calling optional context-oriented methods from within a method taking the context, please use the form:
dao.inX(x).find(id)
rather than
dao.find(x, id);

Reply:
I think this line was missed. It just needs an `inX(x)` and then I think all of the methods have been fixed.